mlkem/poly.c

```diff
@@ -108,13 +109,12 @@ void poly_decompress(poly *r, const uint8_t a[KYBER_POLYCOMPRESSEDBYTES])
 {
     #if (KYBER_POLYCOMPRESSEDBYTES == 128)
     for (int i = 0; i < KYBER_N / 2; i++)
-        __CPROVER_assigns(i, __CPROVER_object_whole(r))
-        __CPROVER_loop_invariant(i >= 0)
-        __CPROVER_loop_invariant(i <= KYBER_N / 2)
-        __CPROVER_loop_invariant(__CPROVER_forall { unsigned int k; (k < i) ==>
-                                 ( r->coeffs[2 * k]     >= 0 && r->coeffs[2 * k]     < KYBER_Q &&
-                                   r->coeffs[2 * k + 1] >= 0 && r->coeffs[2 * k + 1] < KYBER_Q ) })
-        __CPROVER_decreases(KYBER_N / 2 - i)
+        ASSIGNS(i, OBJECT_WHOLE(r))
+        INVARIANT(i >= 0)
+        INVARIANT(i <= KYBER_N / 2)
+        INVARIANT(FORALL(int, k, 0, (i-1), (r->coeffs[2 * k]     >= 0 && r->coeffs[2 * k]     < KYBER_Q &&
+                                            r->coeffs[2 * k + 1] >= 0 && r->coeffs[2 * k + 1] < KYBER_Q)))
```

Comment:
```suggestion
        INVARIANT(FORALL(int, k, 0, 2*(i-1), (r->coeffs[k]     >= 0 && r->coeffs[k]     < KYBER_Q)))
```

Reply:
I see ... let me try that...

Reply:
Yes... works. I also did it using ARRAY_IN_BOUNDS rather than FORALL.

Reply:
Nice!